extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/MapboxCarMapObserver.kt

```diff
@@ -0,0 +1,120 @@
+package com.mapbox.maps.extension.androidauto
+
+import android.graphics.Rect
+import androidx.car.app.SurfaceCallback
+import com.mapbox.maps.EdgeInsets
+import com.mapbox.maps.MapboxExperimental
+import com.mapbox.maps.ScreenCoordinate
+
+/**
+ * Many downstream services will not work until the surface has been created and the map has
+ * loaded. This interface allows you to create custom Mapbox experiences for the car.
+ */
+@MapboxExperimental
+interface MapboxCarMapObserver {
```

Comment:
I like this abstraction 👍

Reply:
since we allow user to add multiple observers to the map, would it cause unexpected behaviour if the order of the onAttach matters? for example loading different style from different observers?

Reply:
yeah the order does matter, and works to a developer's advantage. For example, the route line can [specify a layer](https://docs.mapbox.com/android/navigation/api/2.2.1/libnavui-maps/-u-i%20-maps/com.mapbox.navigation.ui.maps.route.line.model/-mapbox-route-line-options/route-line-below-layer-id.html) that is created by another observer. up to a developer, to organize cross observer dependencies

Reply:
Similar to unit tests https://www.baeldung.com/junit-5-test-order
Except this is, the order you add them to a list. There is not a specified order beyond that